lib/server/controller.js

```diff
@@ -44,6 +48,58 @@ send.mime.define({
   'text/vnd.ga4gh.vcf': ['vcf']
 });
 
+let _parsePostBody = async (request) => {
+  let parsePromise = new Promise((resolve, reject) => {
+    let postTmp;
+    request.on('data', (chunk) => {
+      try {
+        postTmp = jsonLint.parse(chunk,{allowDuplicateObjectKeys: false});
+      } catch (e) {
+        reject("Invalid JSON " + e);
+      }
+      try {
+        schemaValid.validate(postTmp);
+      } catch (e) {
+        reject(e);
+      }
+      resolve(postTmp);
+    });
+    request.on('end', () => {
+      if (typeof postTmp === 'undefined') {
+        reject("No payload in post request");
+      }
+    });
+    request.on('error', () => {
+      reject("Unable to process request");
```

Comment:
Can you add a log entry for the error reported please? info level I think will be good enough.